### PR TITLE
Support different block-type disk drivers

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#7672

#### Additional documentation or context

Deps on https://github.com/longhorn/longhorn-spdk-engine/pull/105

### CODE REVIEW

File: pkg/client/disk.go [hunk 1]
```diff
@@ -162,8 +168,8 @@ func (c *DiskServiceClient) DiskGet(diskType, diskName, diskPath string) (*api.D
 	}, nil
 }
 
-// DiskDelete deletes the disk with the given name and uuid.
-func (c *DiskServiceClient) DiskDelete(diskType, diskName, diskUUID string) error {
+// DiskDelete deletes the disk with the given name, uuid and path and driver.
```
Comment (DamiaSan) [inline pkg/client/disk.go:171]: `// DiskDelete deletes the disk with the given name, uuid, path and driver.`

Comment (derekbit) [reply pkg/client/disk.go]: Updated. Thank you.

File: package/Dockerfile [hunk 2]
```diff
@@ -103,12 +103,13 @@ RUN zypper -n addrepo --refresh https://download.opensuse.org/repositories/syste
...
-    iproute2 qemu-tools e2fsprogs xfsprogs util-linux-systemd python3-pyelftools libcmocka-devel device-mapper netcat && \
+    iproute2 qemu-tools e2fsprogs xfsprogs util-linux-systemd python3-pyelftools libcmocka-devel device-mapper netcat kmod jq util-linux && \
     rm -rf /var/cache/zypp/*
 
 # Install SPDK dependencies
-COPY --from=builder /usr/src/spdk/scripts /spdk/scripts/
-RUN bash /spdk/scripts/pkgdep.sh && rm -rf /spdk
+COPY --from=builder /usr/src/spdk/scripts /usr/src/spdk/scripts
+COPY --from=builder /usr/src/spdk/include /usr/src/spdk/include
+RUN sh /usr/src/spdk/scripts/pkgdep.sh
```
Comment (DamiaSan) [inline package/Dockerfile]: I saw that executing `pkgdep.sh` with `sh` returned some errors, with `bash` instead all was ok

Comment (derekbit) [reply package/Dockerfile]: Yes, so I update it to `base ...`.

**Review by DamiaSan — APPROVED** [review]

**Review by DamiaSan — APPROVED** [review]

**Pull request merged**